### PR TITLE
Adds reviews tests, model, controller, and creation form

### PR DESCRIPTION
**Pull request by efournier92**

### CODE REVIEW

File: app/controllers/apis_controller.rb [hunk 1]
```diff
@@ -9,16 +9,20 @@ def new
...
   def show
     @api = Api.find(params[:id])
+    @review = Review.new
+    @reviews = @api.reviews
+    # @reviews = Reviews.find(params[:id])
   end
 
   def create
     @api = Api.new(api_params)
+    @api.user = User.find(session["warden.user.user.key"][0][0])
```
Comment (radavis) [inline app/controllers/apis_controller.rb]: Devise gives you a `current_user` method. You should use that, instead.

File: app/controllers/reviews_controller.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  def new
+    @review = Review.new
+  end
+
+  def create
+    @api = Api.find(params[:api_id])
+    @review = Review.new(review_params)
+    @review.api = @api
+    @review.rating = params[:review][:rating].to_i
+    @review.user = User.find(session["warden.user.user.key"][0][0])
```
Comment (radavis) [inline app/controllers/reviews_controller.rb]: `current_user`

File: db/seeds.rb [hunk 3]
```diff
@@ -1,13 +1,31 @@
...
+  user: paul
+)
+
+Review.create(
+  title: "Worst Weather API!",
+  body: "It said it would rain, I hate it!",
+  rating: 1,
+  user: paul,
+  api: weather
 )
```
Comment (radavis) [inline db/seeds.rb]: Check out this article on seeding your database, if you haven't already: https://learn.launchacademy.com/teams/spring-2016/curricula/onsite/lessons/seeding

File: spec/features/reviews/user_adds_review_spec.rb [hunk 4]
```diff
@@ -0,0 +1,74 @@
+require "rails_helper"
+
+feature "user adds a new review" do
+  # [x] As a User,
+  #     I want to add a new review for an API
+  #     So others can see and learn from it
+
+  let!(:weather) do
+    Api.create!(
```
Comment (radavis) [inline spec/features/reviews/user_adds_review_spec.rb:9]: You should have a factory for this.

File: app/controllers/apis_controller.rb [hunk 5]
```diff
@@ -9,16 +9,20 @@ def new
 
   def show
     @api = Api.find(params[:id])
+    @review = Review.new
+    @reviews = @api.reviews
+    # @reviews = Reviews.find(params[:id])
```
Comment (alxjrvs) [inline app/controllers/apis_controller.rb]: Please remove Commented out code

File: app/controllers/apis_controller.rb [hunk 6]
```diff
@@ -9,16 +9,20 @@ def new
...
     @api = Api.new(api_params)
+    @api.user = current_user
     if @api.save
       flash[:success] = "New API Created!"
       redirect_to api_path(@api)
     else
-      flash[:error] = @api.errors.full_messages.join(', ')
-      flash[:error] += ". API not created ya bum!"
+      flash[:failure] = @api.errors.full_messages.join(', ')
+      flash[:failure] += ". API not created ya bum!"
```
Comment (alxjrvs) [inline app/controllers/apis_controller.rb:31]: I like the error message, though `error` is the traditional key. Why switch to `failure`?

File: app/controllers/reviews_controller.rb [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+  def index
+    @reviews = Reviews.find(params[:id])
+  end
+
+  def new
+    @review = Review.new
+  end
+
+  def create
+    @api = Api.find(params[:api_id])
```
Comment (alxjrvs) [inline app/controllers/reviews_controller.rb]: instance variables (`@`) are only required when the values are to be consumed in a view. As `create` does not go to a view, consider making these traditional, non-instance variables.

File: app/views/layouts/application.html.erb [hunk 8]
```diff
@@ -4,10 +4,6 @@
     <meta charset="utf-8" />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
 
-    <% flash.each do |name, msg| %>
-      <%= content_tag(:div, msg, class: "alert alert-#{name}") %>
-    <% end %>
-
     <div class=flash_messages>
       <%= display_flash_messages %>
```
Comment (alxjrvs) [inline app/views/layouts/application.html.erb:8]: I'm not familiar with this method! At the risk of sounding obtuse, what does it do?

File: app/controllers/reviews_controller.rb [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+  end
+
+  def create
+    api = Api.find(params[:api_id])
+    review = Review.new(review_params)
+    review.user = current_user
+    review.api = api
+    if review.save
+      flash[:success] = 'Review Submitted!'
+      redirect_to api_path(api)
```
Comment (alxjrvs) [inline app/controllers/reviews_controller.rb:17]: disagree.

File: spec/features/reviews/user_adds_review_spec.rb (hunk 4 above)
Comment (alxjrvs) [inline spec/features/reviews/user_adds_review_spec.rb:9]: agree!

**Pull request merged**